packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNode.ts

```diff
@@ -450,41 +448,18 @@ export class SummarizerNode implements IRootSummarizerNode {
 				if (this.referenceSequenceNumber >= summaryRefSeq) {
 					eventProps.latestSummaryUpdated = false;
 					event.end(eventProps);
-					return { latestSummaryUpdated: false };
+					return { latestSummaryUpdated: false, networkFetchMade: false };
 				}
 
 				// Fetch the latest snapshot and refresh state from it. Note that we need to use the reference sequence number
 				// of the fetched snapshot and not the "summaryRefSeq" that was passed in.
-				const { snapshotTree, snapshotRefSeq: fetchedSnapshotRefSeq } =
-					await fetchLatestSnapshot();
+				await fetchLatestSnapshot();
```

Comment:
Consider doing something like this which removes the logic of fetching snapshot from summarizer node which doesn't need to care:
1. Return { latestSummaryUpdated: true, wasSummaryTracked: false } here. It means that the latest summary was updated by this ack but the summary wasn't tracked. The caller in container runtime then does the work of fetching snapshot and closing.
2. You can remove the `if (this.referenceSequenceNumber >= summaryRefSeq)` block above and do step 1 above in `if (this.referenceSequenceNumber < summaryRefSeq)`

Reply:
You can then remove the `networkFetchMade` property (which is only added for testing) and do the testing based on `latestSummaryUpdated` and `wasSummaryTracked` only.

Reply:
Didn't specifically pull off #2 because I still not exactly sure what you mean here. I will need to do refactor fetchSnapshotFromStorageAndClose and potentially revamp refreshLatestSummary part during the submit summary flow.

Reply:
AB#5417